packages/typespec-ts/src/modular/emitModels.ts

```diff
@@ -132,7 +132,8 @@ export function buildModels(
 ): SourceFile | undefined {
   // We are generating both models and enums here
   const coreClientTypes = new Set<string>();
-  const models = extractModels(codeModel);
+  // filter out the models/enums that are anonymous
+  const models = extractModels(codeModel).filter((m) => !!m.name);
```

Comment:
Filter out anonymous model because we will not generate any models for them.

Reply:
Can you point out to me why the previous filter logic in the below doesn't work ?

Reply:
we prompt the fileter before the length checking to ensure we have items first and then with filter we have 0 items to generate https://github.com/Azure/autorest.typescript/pull/2334/files/e6942358fe894cf50346b3327e4ef05b4ebca043#diff-bd1862ea513d5ada576c700ecf64a8bad65adfe5b04b9bf5c862fffe6589fe72L148-L158.